pennylane/transforms/hamiltonian_expand.py

```diff
@@ -157,6 +157,14 @@ def hamiltonian_expand(tape, group=True):
             tapes.append(new_tape)
 
         def processing_fn(res_groupings):
+            if qml.active_return():
+                res_groupings = [
+                    qml.math.reshape(
+                        qml.math.convert_like(r_group, c_group), qml.math.shape(c_group)
+                    )
+                    for c_group, r_group in zip(coeff_groupings, res_groupings)
```

Comment:
This code introduces a `for` loop that could be merged in the case of the new return types because on lines 168-171 we loop over `res_groupings` again. Does that significantly affect the performance of large Hamiltonians with thousands of terms?

Reply:
Right, I agree.